tests/Microsoft.DotNet.Framework.Docker.Tests/ImageTests.cs

```diff
@@ -121,6 +135,37 @@ private void VerifyImages(ImageDescriptor imageDescriptor, string appDescriptor,
             }
         }
 
+        private void VerifyWCFImages(ImageDescriptor imageDescriptor, string appDescriptor, string runCommand, bool includeRuntime)
+        {
+            List<string> appBuildArgs = new List<string> {  };
+            if (includeRuntime)
+            {
+                string baseRuntimeImage = GetImage("wcf", imageDescriptor.RuntimeVersion, imageDescriptor.OsVariant);
+                appBuildArgs.Add($"BASE_RUNTIME_IMAGE={baseRuntimeImage}");
+            }
+
+            string appId = $"{appDescriptor}-{DateTime.Now.ToFileTime()}";
+            string workDir = Path.Combine(
+                Directory.GetCurrentDirectory(),
+                "projects",
+                $"{appDescriptor}-{imageDescriptor.RuntimeVersion}");
+
+            try
+            {
+                DockerHelper.Build(
+                    tag: appId,
+                    dockerfile: Path.Combine(workDir, "Dockerfile"),
+                    buildContextPath: workDir,
+                    buildArgs: appBuildArgs);
+
+                DockerHelper.Run(image: appId, containerName: appId, command: runCommand);
```

Comment:
@HongGit ?

Reply:
I thought this approach match the pattern of existing dotnet tests.  Since ASP.NET and WCF images are running container images, this approach will shut down the container after sending out the request.  This is also simple, and reliable.  Do we have to use the pattern suggested above?

Reply:
The same pattern doesn't have to be used.  However I have concerns with the proposed pattern

1. It doesn't feel like a canonical Dockerfile at all.  We should be testing in a way we expect the product to be used.
1. I have concerns about the reliability of this pattern.  The hard coded wait in the script doesn't provide the flexibility to retry.
1. The maintenance of having the test code copied in each project is not ideal.

Reply:
For #1, I am not sure that I understand your point, this is a simple way how the product is used.  For WCF web service, .svc file needs to be deployed to the web server, and the url is used to generate the proxy.  
For #2, the reason there is a sleep is because IIS web server needs to be warmed up after start.  Are you suggesting to add retry if it fails at the beginning?
For #3, can you please elaborate this a little bit?  Is this how the dotnet tests are?

Reply:
1. Accessing the service within the WCF container itself is not the canonical usage to me?  I would expect a client to be involved.  How is the test validating the service is externally accessible?
2. From my experience, having a sleep like this hard coded is going to be problematic.  It would be more robust to have test logic that will retry accessing the service a number of times after a fixed sleep time.
3.  The subtle difference is the separation of the test projects and the test code.  These test projects have the logic to "test" the service embedded into the EntryPoint.ps1 script.  This common test code better belongs IMO within the WCF test case itself.

Reply:
updated.